src/vehicle_gui.cpp

```diff
@@ -1373,8 +1373,8 @@ static bool VehicleCargoSorter(const Vehicle * const &a, const Vehicle * const &
 	for (v = b; v != nullptr; v = v->Next()) diff[v->cargo_type] -= v->cargo_cap;
 
 	int r = 0;
-	for (CargoID i = 0; i < NUM_CARGO; i++) {
-		r = diff[i];
+	for (uint d : diff) {
+		r = d;
 		if (r != 0) break;
 	}
```

Comment:
Couldn't this be something along the line of the following?
```
auto it = std::find_if(diff.begin(), diff.end(), [](auto d) { return d != 0; });
return it != diff.end() ? *it < 0 : VehicleNumberSorter(a, b);
```
It's sort of a recurring pattern with the capacities as well, though there the code is smaller than it's here.